Servers/UI/OJS.Servers.Ui/ClientApp/src/components/modal/ContestModal.tsx

```diff
@@ -1,41 +1,41 @@
-import React, { useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import Box from '@mui/material/Box';
 import Modal from '@mui/material/Modal';
 import Typography from '@mui/material/Typography';
 
-import { IContestModal } from '../../common/types';
+import { IContestModalInfoType } from '../../common/types';
 import { useCurrentContest } from '../../hooks/use-current-contest';
 import { getHomePageUrl } from '../../utils/urls';
 import Button, { ButtonSize, ButtonType } from '../guidelines/buttons/Button';
 
 import styles from './ContestModal.module.scss';
 
 interface IContestModalProps {
-    contest: IContestModal;
-    isShowing: boolean;
-    toggle: () => void;
+    contest: IContestModalInfoType;
 }
 
-const ContestModal = ({ contest, isShowing, toggle }: IContestModalProps) => {
+const defaultState = { state: { isShowing: true } };
+
+const ContestModal = ({ contest }: IContestModalProps) => {
```

Comment:
Don't overcomplicate the elements in the modal. Replacing `Box` and `Typography` components that you use from material UI does not give extraordinary benefits rather than using div and span and applying simple css styling.

Also you are using these strings on 2 places. Move them into constants, don't repeat them.
![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/19411274/fa86d335-ce9b-469d-a85c-73ac36615274)

Reply:
Done.